### PR TITLE
fix(mcp): reject control characters in FinDrive filenames

### LINKED ISSUE #360: Bug_151_MUST_FIX: FD-FNAME-003 — Newline character in filename enables log injection
**Component:** finbot/mcp/servers/findrive/server.py → upload_file
**Root cause:**
``` python
f = repo.create_file(filename=filename, ...)
# no control character filtering
```
**Steps to reproduce:**
1. Call upload_file(filename='legit.pdf\nFAKE LOG ENTRY: admin login from 1.2.3.4', ...)

**Expected:** error — control characters rejected
**Actual:** file created; newline stored verbatim in filename

**How to execute:**
``` bash
pytest tests/unit/mcp/test_findrive.py::TestFileNameValidation::test_fd_fname_003_newline_in_filename_enables_log_injection -v
```
**Proposed fix:** Add control character check in upload_file:
``` python
if any(c in filename for c in ('\n', '\r', '\t', '\x00')):
    return {"error": "filename contains invalid control characters"}
```
**Impact:** Newlines in filenames inject fake entries into application logs, corrupting audit trails and deceiving log monitoring.

**Acceptance criteria:**
* test_fd_fname_003_newline_in_filename_enables_log_injection passes (newline rejected)
* test_fd_upload_001_upload_returns_file_id_and_metadata continues to pass (valid filenames still work)

### PR DESCRIPTION
**Pull request by Deez-Automations**
## Summary

Fixes #360.

`upload_file` (FinDrive MCP server) passed `filename` straight through to storage and into a log line with no validation at all. A filename containing a newline is stored verbatim in the DB and, more importantly, injects a fake log line into the application log — corrupting the audit trail and potentially deceiving anyone monitoring logs for suspicious activity.

## Fix

Rejects `\n`, `\r`, `\t`, and `\x00` in filenames before they reach storage or logging, matching the fix proposed in the issue. Ordinary filenames are unaffected.

## Test plan

- [x] New test file `tests/unit/mcp/test_findrive.py` — reproduces the vulnerability first (confirmed failing against the unfixed code), then confirms the fix
- [x] Covers newline, carriage return, and null byte injection
- [x] Regression test confirms ordinary valid filenames continue to work unaffected
- [x] `pytest tests/unit/mcp/test_findrive.py` — 4/4 passing

### CODE REVIEW

**Comment by Deez-Automations** [comment]
Addressed the Copilot review feedback:

- **Test naming**: kept `test_fd_fname_003_newline_in_filename_enables_log_injection` as-is since it matches issue #360's own specified acceptance-criteria test name verbatim (useful for traceability back to the issue), but added a docstring note clarifying the name describes the vulnerability being guarded against, not the assertion direction.
- **Missing tab-character coverage**: added a dedicated test — the fix already rejected `\t`, but the original suite only covered newline, carriage return, and null byte.

5/5 tests passing after both changes.

File: tests/unit/mcp/test_findrive.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    return tool.fn(**kwargs)
+
+
+class TestFileNameValidation:
+
+    @pytest.mark.unit
+    @pytest.mark.asyncio
+    async def test_fd_fname_003_newline_in_filename_enables_log_injection(
+        self, db, session_context
+    ):
```
[the remark this answers is not in the document]
Comment (Deez-Automations) [reply tests/unit/mcp/test_findrive.py:38]: Kept the name as-is on purpose — it matches issue #360's own acceptance-criteria test name verbatim so it's traceable back to the ticket, added a docstring note explaining that so it's not confusing on its own.

File: tests/unit/mcp/test_findrive.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            session_context,
+            filename="legit.pdf\rFAKE LOG ENTRY",
+            content="some content",
+        )
+
+        assert "error" in result
+
+    @pytest.mark.unit
+    @pytest.mark.asyncio
+    async def test_fd_fname_003c_null_byte_in_filename_rejected(self, db, session_context):
```
[the remark this answers is not in the document]
Comment (Deez-Automations) [reply tests/unit/mcp/test_findrive.py:67]: Already added — test_fd_fname_003d_tab_in_filename_rejected covers it.
